### PR TITLE
HIVE-25416 : Hive metastore memory leak because datanucleus-api-jdo bug

### PR DESCRIPTION
**Pull request by zzzzming95**
### What changes were proposed in this pull request?

Upgrade datanucleus-api-jdo version to 5.2.6 to fix metostore memory leak bug.

### Why are the changes needed?

Metostore metostore memory leak will cause the service to be unavailable.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

can refer to  https://github.com/datanucleus/datanucleus-api-jdo/issues/106.

That is killed some process in mysql like:
```
mysql>show processlist;
mysql> kill "number from first col";
```

### CODE REVIEW

**Review by kgyrtkirk — COMMENTED** [review]
sorry for the late response...
do you think there are any obstacles to go up to 5.2.7?
https://mvnrepository.com/artifact/org.datanucleus/datanucleus-api-jdo/5.2.7

**Comment by shezhiming** [comment]
@kgyrtkirk 

It seems that there is no obstacle.

**Review by slachiewicz — APPROVED** [review]

**Comment by zzzzming95** [comment]
@kasakrisz  @marton-bod @kgyrtkirk @pvary 

CI build fails not because of this pr. Can you guys merge this pr into master , I think it's a necessary patch.

**Review by slachiewicz — APPROVED** [review]

**Review by pvary — APPROVED** [review]

**Comment by kgyrtkirk** [comment]
try to upgrade datanucleus to the latest - not to 5.2.6
https://mvnrepository.com/artifact/org.datanucleus/datanucleus-api-jdo

please update the PR

a successfull CI run is required before merging changes - no exceptions.

**Comment by pvary** [comment]
I missed that 5.2.6 is not the latest. This case I agree with @kgyrtkirk that we should try at least 5.2.8, and use older one only if we find some issues.

We have a green run BTW 😄

File: pom.xml [hunk 1]
```diff
@@ -108,10 +108,10 @@
     <avro.version>1.8.2</avro.version>
     <bcprov-jdk15on.version>1.64</bcprov-jdk15on.version>
     <calcite.version>1.25.0</calcite.version>
-    <datanucleus-api-jdo.version>5.2.4</datanucleus-api-jdo.version>
-    <datanucleus-core.version>5.2.4</datanucleus-core.version>
+    <datanucleus-api-jdo.version>5.2.6</datanucleus-api-jdo.version>
```
Comment (nrg4878) [inline pom.xml:111]: According to the release notes, this memory leak issue is fixed in 5.2.7 and NOT in 5.2.6. 
https://www.datanucleus.org/documentation/news/access_platform_5_2.html#_datanucleus_accessplatform_5_2_7
So this PR will not resolve the issue.

Also there are some important bug fixes like support for sequences in MS SQL etc in later minor releases. We should atleast try to move up to 5.2.10 (5.2.11 is also available, but not many fixes that directly affect hive).

**Review by nrg4878 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (zzzzming95) [reply pom.xml]: @nrg4878 
This was my oversight, now I have upgraded the various packages to the latest version.

[datanucleus-core](https://mvnrepository.com/artifact/org.datanucleus/datanucleus-core)
[datanucleus-api-jdo](https://mvnrepository.com/artifact/org.datanucleus/datanucleus-api-jdo)
[datanucleus-rdbms](https://mvnrepository.com/artifact/org.datanucleus/datanucleus-rdbms)

@pvary @kgyrtkirk Please review again.
thanks all guys~

Comment (nrg4878) [reply pom.xml]: Thanks @zzzzming95  for addressing the issue.

**Review by nrg4878 — APPROVED** [review]
Change looks good to me. +1

**Comment by pvary** [comment]
@nrg4878: Is it intentional that we left HIVE-25416 open?

**Comment by zzzzming95** [comment]
> @nrg4878: Is it intentional that we left [HIVE-25416](https://issues.apache.org/jira/browse/HIVE-25416) open?

i

> @nrg4878: Is it intentional that we left [HIVE-25416](https://issues.apache.org/jira/browse/HIVE-25416) open?

sorry , I forget to change the status , now i make it as RESOLVED.

**Pull request merged**